# Домашнє завдання до теми "Основи роботи з рядками та математичні операції"

1. Створіть змінні `a`, `b`, `c` та `d`, задайте їм значення відповідно 101, 210, 111, 402, та обчисліть з допомогою Python наступний вираз:  $$ \frac{(a+b) * c}{d} $$

**Важливо**: тут ми маємо обчислити значення саме з використанням змінних, а не просто чисел, як би ми робили у калькуряторі. Змінні потрібні аби за бажання ми могли змінити їх значення та не переписувати код обчислення виразу, бо він буде працювати для різних значень змінних.

In [20]:
a = 101
b = 210
c = 111
d = 402
e = (a + b) * c / d
e

85.8731343283582

Очікувана відповідь: 85.8731343283582

2. Якою буде остача від ділення числа 11283 на 89? Обчисліть з допомогою Python.

In [22]:
11283 % 89

69

In [24]:
#АБО
a = 11283
b = 89
c = a % b
c

69

3. Скільки цілих разів поміститься число 89 в 11283? (Або що те саме: який буде результат цілочисельного ділення 11283 на 89?)

In [26]:
11283 // 89

126

In [30]:
#АБО
d = a // b
d

126

4. Запишіть у змінну `name` ваше імʼя.

In [32]:
name = 'Yulianna'

5. Запишіть у змінну `surname` ваше прізвище.

In [34]:
surname = 'Tereshko'

6. Сконкатенуйте імʼя та прізвище в одну фразу через пробіл. Запишіть результат у нову змінну `name_surname` та виведіть результат на екран.  

In [44]:
name_surname = name + ' ' + surname
name_surname

'Yulianna Tereshko'

Очікувана відповідь (тільки з вашими даними): 'Hanna Pylieva'

7. Зробіть усі літери змінної `name_surname` заглавними використовуючи метод рядків в Python:

In [46]:
name_surname.upper()

'YULIANNA TERESHKO'

Очікувана відповідь (тільки з вашими даними): 'HANNA PYLIEVA'

8. Виведіть третю літеру вашого `name_surname`. Тут уважно, треба саме третя літера. Згадайте, з якого значення починається індексування значень в Python.

In [48]:
name_surname [2]

'l'

Очікувана відповідь (тільки з вашими даними): 'n'

9. Виведіть першу та останню літеру вашого прізвища.

In [164]:
surname [0], surname [len (surname) - 1]

('T', 'o')

коментар: спочатку я написала ось так: surname [0], surname [7], але потім подумала, що це не зовсім правильний розв"язок, адже я можна сказати "підлаштувала" розв"язок під умову конкретної задачі. Тобто порахувала кількість символів у змінній surname і просто вписала номер останнього. І вирішила спробувати зробити більш універсальний розв"язок, який підходив би під змінну з іншою довжиною без потреби рахувати вручну кількість символів перед написанням коду)) і в результаті вийшло surname [len (surname) - 1]

Очікувана відповідь (тільки з вашими даними): ('P', 'a')

10. Виведіть довжину вашого прізвища.

In [166]:
len(surname)

8

Очікувана відповідь (тільки з вашими даними): 7


11. Виведіть кількість літер `a` у столиці Мадагаскару `Antananarivo`. Важливо врахувати всі літери, як великі, так і маленькі.

In [168]:
text = "Antananarivo"
text1 = text.lower()
x = text1.count('a')
print(x)

4


задача, яка зайняла найбільше часу. Я одразу подумала, що має бути якийсь метод count, але дуже довго не могла знайти, як саме його застосовувати))

Очікувана відповідь: 4

12. Створіть рядок `my_string` зі значенням: "The quick brown fox jumps over the lazy dog!!!".   
Перевірте, чи починається рядок `my_string` з "The" і чи закінчується він "dog!". Виведіть результати на екран.

In [170]:
my_string = "The quick brown fox jumps over the lazy dog!!!"
print (my_string.startswith("The"))
print (my_string.endswith("dog!"))

True
False


13. Розділіть рядок `my_string` на список окремих слів, а потім з'єднайте всі слова через нижнє підкреслювання. Виведіть результат на екран.

In [150]:
my_string.split(' ')

['The', 'quick', 'brown', 'fox', 'jumps', 'over', 'the', 'lazy', 'dog!!!']

In [154]:
'_'.join(['The', 'quick', 'brown', 'fox', 'jumps', 'over', 'the', 'lazy', 'dog!!!'])

'The_quick_brown_fox_jumps_over_the_lazy_dog!!!'

Очікуваний результат: `The_quick_brown_fox_jumps_over_the_lazy_dog!!!`